src/DGtal/shapes/SurfaceMeshHelper.h

```diff
@@ -0,0 +1,454 @@
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file SurfaceMeshHelper.h
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2020/07/11
+ *
+ * Header file for module SurfaceMeshHelper.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(SurfaceMeshHelper_RECURSES)
+#error Recursive header files inclusion detected in SurfaceMeshHelper.h
+#else // defined(SurfaceMeshHelper_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define SurfaceMeshHelper_RECURSES
+
+#if !defined SurfaceMeshHelper_h
+/** Prevents repeated inclusion of headers. */
+#define SurfaceMeshHelper_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <sstream>
+#include <string>
+#include "DGtal/base/Common.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/shapes/SurfaceMesh.h"
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class SurfaceMeshHelper
+  /**
+     Description of template class 'SurfaceMeshHelper' <p> \brief Aim:
+     An helper class for building classical meshes.
+
+     @tparam TRealPoint an arbitrary model of RealPoint.
+     @tparam TRealVector an arbitrary model of RealVector.
+  */
+  template < typename TRealPoint, typename TRealVector >
+  struct SurfaceMeshHelper
+  {
+    typedef TRealPoint                              RealPoint;
+    typedef TRealVector                             RealVector;
+    typedef SurfaceMeshHelper< RealPoint, RealVector > Self;
+    static const Dimension dimension = RealPoint::dimension;
+    BOOST_STATIC_ASSERT( ( dimension == 3 ) );
+
+    typedef DGtal::SurfaceMesh< RealPoint, RealVector > SurfaceMesh;
+    typedef typename RealVector::Component          Scalar;
+    typedef std::vector<Scalar>                     Scalars;
+    typedef std::vector<RealVector>                 RealVectors;
+    typedef typename SurfaceMesh::Size           Size;
+    typedef typename SurfaceMesh::Index          Index;
+    typedef typename SurfaceMesh::Vertices       Vertices;
+    typedef typename SurfaceMesh::Faces          Faces;
+
+    /// Enum type for specifying if the mesh should have after
+    /// construction: NO_NORMALS no normals, VERTEX_NORMALS the
+    /// normals at each vertex, FACE_NORMALS the normals at each face.
+    enum class Normals { NO_NORMALS, VERTEX_NORMALS, FACE_NORMALS };
```

Comment:
I was confused between Normals as normal container (aka std::vector<RealVector> for instance, following previous convention) and this tag..

I would rename this as "NormalType/Tag" or something like that